jdtls.ext/com.shengchen.checkstyle.runner/src/main/java/com/shengchen/checkstyle/quickfix/whitepace/NewlineAtEndOfFileQuickFix.java

```diff
@@ -0,0 +1,45 @@
+/*
+ * Copyright (C) jdneo
+
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * any later version.
+
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package com.shengchen.checkstyle.quickfix.whitepace;
+
+import com.shengchen.checkstyle.quickfix.BaseEditQuickFix;
+
+import org.eclipse.jface.text.BadLocationException;
+import org.eclipse.jface.text.Document;
+import org.eclipse.jface.text.IRegion;
+import org.eclipse.text.edits.InsertEdit;
+import org.eclipse.text.edits.TextEdit;
+
+public class NewlineAtEndOfFileQuickFix extends BaseEditQuickFix {
+
+    @Override
+    public TextEdit createTextEdit(IRegion lineInfo, int markerStartOffset, String violationKey, Document doc) {
+        try {
+            final String lastChar = doc.getLength() > 0 ? doc.get(doc.getLength() - 1, 1) : "";
+            
+            if (lastChar.isEmpty() || !"\n".equals(lastChar)) {
+                return new InsertEdit(doc.getLength(), "\n");
```

Comment:
`\n` might not be enough. Given this rule has the options:

`<property name="lineSeparator" value="lf"/>`

Reply:
@jdneo hmm, yes; given the error that is reported doesn't specify what it's looking for, I think the best we can do is assume the default (which looks for any of CR LF or CRLF) and then adds the platform-specific version.

Or we could simply add the platform specific version.

Is it handy enough to fix these things that it might be wrong sometimes? Or should we more actively pursue getting Checkstyle to expose its internal configuration? My forays into the code suggest that it doesn't want to.